src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/KeyValuePairListEncoding.cs

```diff
@@ -76,12 +76,13 @@ static string Decode(string value)
         }
     }
 
-    public static string Format(IEnumerable<(string Name, string Value)> pairs)
+    public static string Format(IEnumerable<(string Name, string Value)> pairs, string separator = ",")
     {
         // Copied from ActiveLaunchProfileEnvironmentVariableValueProvider in the .NET Project System.
         // In future, EnvironmentVariablesNameValueListEncoding should be exported from that code base and imported here.
-
-        return string.Join(",", pairs.Select(kvp => $"{Encode(kvp.Name)}={Encode(kvp.Value)}"));
+        return string.Join(separator, pairs.Select(kvp => string.IsNullOrEmpty(kvp.Value) 
+            ? Encode(kvp.Name) 
+            : $"{Encode(kvp.Name)}={Encode(kvp.Value)}"));
```

Comment:
There's a lot going on here that could be made cleare with some more wrapping. I think the most common form used in this repo would be:

```suggestion
        return string.Join(
            separator,
            pairs.Select(kvp => string.IsNullOrEmpty(kvp.Value) 
                ? Encode(kvp.Name) 
                : $"{Encode(kvp.Name)}={Encode(kvp.Value)}"));
```

Reply:
changed